octodns_bind/__init__.py

```diff
@@ -97,27 +97,25 @@ def __init__(self, id, directory, file_extension='.', check_origin=True):
     def _load_zone_file(self, zone_name):
         zone_filename = f'{zone_name[:-1]}{self.file_extension}'
         zonefiles = listdir(self.directory)
+        path = join(self.directory, zone_filename)
         if zone_filename in zonefiles:
             try:
                 z = dns.zone.from_file(
-                    join(self.directory, zone_filename),
+                    path,
                     zone_name,
                     relativize=False,
                     check_origin=self.check_origin,
                 )
             except DNSException as error:
                 raise ZoneFileSourceLoadFailure(error)
         else:
-            raise ZoneFileSourceNotFound()
+            raise ZoneFileSourceNotFound(path)
 
         return z
 
     def zone_records(self, zone):
         if zone.name not in self._zone_records:
-            try:
```

Comment:
So this is a change in behavior, I'm unsure if this is okay or we should preserve existing behavior and just pop up a warning.

Previously if the file is not found, it will be caught and just return `[]` and continue on. For a user to know the zone file isn't found they'd have to do some digging.

With this change it will raise an exception and stop the run which is somewhat consistent with the AxfrSource when it can't complete a zone transfer

Reply:
I think this is preferable and it should be fine to error out in an undesirable case. I'd only be worried about the change in behavior if it could lead to making unexpected changes or something of that nature. I'm fine with things blowing up and telling you to fix them (when it previously didn't.)